google/cloud/spanner/internal/partial_result_set_reader.cc

```diff
@@ -115,8 +120,36 @@ Status PartialResultSetReader::ReadFromStream() {
   }
 
   // TODO(#271) store and use resume_token.
-  // TODO(#270) handle chunked_value.
+
   values_.Swap(result_set.mutable_values());
+  // If the last response had `chunked_value` set, partial_chunked_value_
+  // contains the partial value; merge it with the first value in the
+  // response.
+  if (partial_chunked_value_.has_value()) {
+    if (values_.empty()) {
+      return Status(StatusCode::kInternal,
+                    "PartialResultSet contained no values to merge with prior "
+                    "chunked_value");
+    }
+    auto merge_status =
+        MergeChunk(*partial_chunked_value_, std::move(values_[0]));
```

Comment:
Values can be chunked into *multiple* subsequent Values ([example](https://github.com/googleapis/googleapis/blob/11592a15391951348a64f5c303399733b1c5b3b2/google/spanner/v1/result_set.proto#L117-L135)). So I think we need to keep checking the `chunked_value` bool and keep merging in a loop until we get one that is not chunked.

Reply:
Just curious, in that example the intermediate [value](https://github.com/googleapis/googleapis/blob/11592a15391951348a64f5c303399733b1c5b3b2/google/spanner/v1/result_set.proto#L127-L131) has only one element, is it possible for it to have multiple elements?

In other words, how does one interpret something like (not that I split the value in the second entry):

```
  //     {
  //       "metadata": { ... }
  //       "values": ["Hello", "W"]
  //       "chunked_value": true
  //       "resume_token": "Af65..."
  //     }
  //     {
  //       "values": ["o", "r", "l"]
  //       "chunked_value": true
  //       "resume_token": "Bqp2..."
  //     }
  //     {
  //       "values": ["d"]
  //       "resume_token": "Zx1B..."
  //     }
```